kubernetes-client/src/main/java/io/micronaut/kubernetes/client/DiscoveryCache.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client;
+
+import io.kubernetes.client.Discovery;
+import io.kubernetes.client.apimachinery.GroupVersionKind;
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.openapi.ApiException;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.annotation.Value;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Bean that provides caching over the {@link Discovery} results.
+ *
+ * @author Pavol Gressa
+ * @since 3.1
+ */
+@Requires(beans = Discovery.class)
+@Singleton
+public class DiscoveryCache {
```

Comment:
Note: This along with DiscoveryFactory provides a way how to get Api resource details like whether the resource is namespaced, api group, resourceplural.. basically it allows us to have such simple declaration of informer like: `@Informer(apiResoruceType = ... , apiResourceListType = ... )`

Reply:
Perhaps annotate this with `@Internal`? Does it need to be public api?